qiita_ware/util.py

```diff
@@ -117,104 +116,14 @@ def stats_from_df(df):
     out = {}
 
     for column in natsorted(df.columns):
+        if column in {'study_id'}:
+            continue
         counts = df[column].value_counts()
 
         # get a pandas series of the value-count pairs
-        out[column] = [(key, counts[key]) for key in natsorted(counts.index)]
-
-    return out
-
-
-def metadata_stats_from_sample_and_prep_templates(st, pt):
-    """Create a dictionary of summary statistics for the sample/prep templates
-
-    Parameters
-    ----------
-    st : SampleTemplate
-        Initialized SampleTemplate to use for the metadat stats.
-    pt : PrepTemplate
-        Initialized PrepTemplate to use for the metadat stats.
-
-    Returns
-    -------
-    dict
-        Dictionary object where the keys are the names of the metadata
-        categories and the keys are tuples where the first element is the name
-        of a metadata value in category and the second element is the number of
-        times that value was seen.
-    """
-    df = metadata_map_from_sample_and_prep_templates(st, pt)
-    return stats_from_df(df)
+        out[str(column)] = [(str(key), counts[key])
+                            for key in natsorted(counts.index)]
 
-
-def metadata_map_from_sample_and_prep_templates(st, pt):
-    """Create a mapping file from a sample and a prep template
-
-    Parameters
-    ----------
-    st : SampleTemplate
-        Initialized SampleTemplate to use.
-    pt : PrepTemplate
-        Initialized PrepTemplate to use.
-
-    Returns
-    -------
-    pd.DataFrame
-        A DataFrame object where the index values are the sample identifiers
-        and the column names are the metadata categories.
-
-    Notes
-    -----
-    If the sample and preparation templates include repeated column names these
-    will be repeated in the resulting object, as they are not supposed to have
-    overlapping column names.
-    """
-    s_df = dataframe_from_template(st)
-    p_df = dataframe_from_template(pt)
-
-    return pd.merge(s_df, p_df, left_index=True, right_index=True, how='outer')
-
-
-def dataframe_from_template(t):
```

Comment:
There are prep template summary and sample template summary links, but no qiime mapping file summary file link. I think this is a remnant of an idea that never got deleted once it moved on, since nothing calls it.